### PR TITLE
Rov valor buffs

### PR DESCRIPTION
**Pull request by kaincenteno**
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Changes made are for oneventstart to show correct price and
oneventfinish to charge the new lower price.

Dunno why its also commiting old commits? (i just forked this)

### CODE REVIEW

**Comment by kaincenteno** [comment]
any soul willing to review this? all i did was just add an if statement to check for a param that decides if u have key item or not, then copied one of the tables but with the newer price deducted if u have keyitem ^^;;

**Comment by cocosolos** [comment]
I've looked through this and I believe it works. I'm not sure if we should duplicate the table just for a different price, or if it would be better for the table to return functions, something like this:
```lua
[117] = function (rhapsodyInWhite) if rhapsodyInWhite then return {act = "DRIED_MEAT",      cost = 25, food = true} else return {act = "DRIED_MEAT",      cost = 50, food = true} end end,
```
or
```lua
[117] = function (rhapsodyInWhite) return {act = "DRIED_MEAT",      cost = function() if rhapsodyInWhite then return 25 else return 50 end end, food = true} end,
```
then check the key item like this
```lua
    for k,v in pairs(regimeInfo[regimeType].finishOptions) do
        out[k] = v(hasKI)
```
So I'll let someone chime in on that otherwise I have no problems merging.

**Comment by TeoTwawki** [comment]
another possibility is to leave it one table and leave cost alone, but then add another entry for the RoV modified cost like
`[ 21] = {act = "REPATRIATION",    cost = 50, discounted = 10},`
and then change the for loop part to check if you have KI before deciding which one to use in the loop.

I'd go with the functions coco suggested if thats not easy to do.

**Comment by cocosolos** [comment]
> `[ 21] = {act = "REPATRIATION", cost = 50, discounted = 10},`

I like this, much cleaner. Would have to update 2 places in `bookOnEventFinish` to check for KI and use either `opt.cost` or `opt.discounted`.

**Comment by kaincenteno** [comment]
@cocosolos done :cat: 
That SQL file came back again >.< can you remove it please

File: scripts/globals/regimes.lua [hunk 1]
```diff
@@ -1050,21 +1057,27 @@ tpz.regime.bookOnEventFinish = function(player, option, regimeType)
...
-        return
+    if hasKI then
+        if opt.discounted and opt.discounted > tabs then
+            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
+            return
+        elseif opt.cost and opt.cost > tabs then
+            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
+            return
+        end
     end
```
Comment (cocosolos) [inline scripts/globals/regimes.lua]: Should this be checking regular cost if no KI? Like this:
```lua
    if hasKI then
        if opt.discounted and opt.discounted > tabs then
            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
            return
        end
    else
        if opt.cost and opt.cost > tabs then
            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
            return
        end
    end
```

Comment (kaincenteno) [reply scripts/globals/regimes.lua]: fixed :)

**Review by cocosolos — APPROVED** [review]
Looks good, thank you!

**Review by ibm2431 — COMMENTED** [review]
```lua
local cost = opt.cost

if cost and player:hasKeyItem(tpz.ki.RHAPSODY_IN_WHITE)
    cost = opt.discounted
end

if cost > tabs then
    -- Not enough
end

if opt.act then
    player:delCurrency("valor_point", cost)
```

**Comment by kaincenteno** [comment]
it's the curse of the 666 

changes done.

**Review by ibm2431 — APPROVED** [review]

**Pull request merged**